resources/js/beatmap-discussions/nominations.tsx

```diff
@@ -0,0 +1,680 @@
+    const discussions = Object.values(this.props.currentDiscussions.byFilter.hype.generalAll);
+    const userAlreadyHyped = currentUser != null && discussions.some((discussion) => discussion?.user_id === currentUser.id);
+
+    return (
+      <BigButton
+        disabled={!this.props.beatmapset.current_user_attributes.can_hype}
+        icon='fas fa-bullhorn'
+        modifiers='warning'
+        props={{
+          onClick: this.focusHypeInput,
+          title: this.props.beatmapset.current_user_attributes.can_hype_reason,
+        }}
+        text={userAlreadyHyped ? trans('beatmaps.hype.button_done') : trans('beatmaps.hype.button')}
+      />
+    );
+  }
+
+  private renderLightsForNominations(nominations?: BeatmapsetNominationsInterface) {
+    if (nominations == null) return null;
+
+    const hybrid = Object.keys(this.props.beatmapset.nominations.required).length > 1;
+
+    return (
+      <div className={classWithModifiers(`${bn}__discrete-bar-group`, { hybrid })}>
+        {nominations.legacy_mode ? (
+          <DiscreteBar current={nominations.current} total={nominations.required} />
+        ) : Object.entries(nominations.required).map(([ruleset, required]: [GameMode, number]) => (
+          <DiscreteBar
+            key={ruleset}
+            current={nominations.current[ruleset] ?? 0}
+            label={hybrid ? <i className={`fal fa-extra-mode-${ruleset}`} /> : null}
+            modifiers={{ 'beatmapset-nomination-hybrid' : hybrid }}
+            total={required}
+          />
+        ))}
+      </div>
+    );
+  }
+
+  private renderLoveBeatmapDialog() {
+    if (!this.showLoveBeatmapDialog) return null;
+
+    return (
+      <Modal>
+        <LoveBeatmapDialog
+          beatmapset={this.props.beatmapset}
+          onClose={this.handleToggleLoveBeatmapDialog}
+        />
+      </Modal>
+    );
+  }
+
+  private renderLoveButton() {
+    if (!this.props.beatmapset.current_user_attributes.can_love) return null;
+
+    return (
+      <BigButton
+        icon='fas fa-heart'
+        modifiers='pink'
+        props={{
+          onClick: this.handleToggleLoveBeatmapDialog,
+        }}
+        text={trans('beatmaps.nominations.love')}
+      />
+    );
+  }
+
+  private renderNominationBar() {
+    const requiredHype = this.props.beatmapset.hype?.required ?? 0; // TODO: skip if null?
+    const hypeRaw = this.props.currentDiscussions.totalHype;
+    const mapCanBeNominated = this.props.beatmapset.status === 'pending' && hypeRaw >= requiredHype;
+
+    if (!(mapCanBeNominated || this.isQualified)) return null;
+
+    const nominations = this.props.beatmapset.nominations;
+
+    return (
+      <div>
+        <div className={`${bn}__header`}>
+          <span className={`${bn}__title`}>{trans('beatmaps.nominations.title')}</span>
+          <span>{` ${nominationsCount(nominations, 'current')} / ${nominationsCount(nominations, 'required')}`}</span>
```

Comment:
the space prefix doesn't do anything, it's a bit shorter with just jsx syntax, and it can use some number formatting.